compaction/compaction_manager.cc

```diff
@@ -1524,23 +1524,27 @@ class cleanup_sstables_compaction_task_executor : public compaction_task_executo
         co_return std::nullopt;
     }
 private:
-    // Releases reference to cleaned files such that respective used disk space can be freed.
-    void release_exhausted(std::vector<sstables::shared_sstable> exhausted_sstables) {
-        _compacting.release_compacting(exhausted_sstables);
-    }
-
     future<> run_cleanup_job(sstables::compaction_descriptor descriptor) {
         co_await coroutine::switch_to(_cm.compaction_sg().cpu);
 
+        // Releases reference to cleaned files such that respective used disk space can be freed.
+        auto release_exhausted = [this, &descriptor] (std::vector<sstables::shared_sstable> exhausted_sstables) mutable {
+            auto exhausted = boost::copy_range<std::unordered_set<sstables::shared_sstable>>(exhausted_sstables);
+            auto e = boost::range::remove_if(descriptor.sstables, [&] (const sstables::shared_sstable& sst) {
```

Comment:
why not use std::erase_if?
https://en.cppreference.com/w/cpp/container/vector/erase2

Reply:
I am stuck with older C++ way of erasing a set of elements from a vector. thanks for the suggestion, will switch to it in v2.